internal/k8s/builder/builder.go

```diff
@@ -31,7 +31,7 @@ func init() {
 }
 
 const (
-	defaultEnvoyImage           = "envoyproxy/envoy:v1.21-latest"
```

Comment:
honestly I'm not a fan of making this change here, for some reason the envoy image specified in the e2e tests don't actually override this and I'm not 100% sure on what else I would need to change